src/binding/python/RecordComponent.cpp

```diff
@@ -29,13 +29,155 @@
+ */
+inline py::array
+load_chunk(RecordComponent & r, Offset const & offset, Extent const & extent)
+{
+    std::vector<ptrdiff_t> shape(extent.size());
+    std::copy(std::begin(extent), std::end(extent), std::begin(shape));
+
+    auto const dtype = dtype_to_numpy( r.getDatatype() );
+
+    auto a = py::array( dtype, shape );
+
+    if( r.getDatatype() == Datatype::CHAR )
+        r.loadChunk<char>(shareRaw((char*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::UCHAR )
+        r.loadChunk<unsigned char>(shareRaw((unsigned char*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::SHORT )
+        r.loadChunk<short>(shareRaw((short*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::INT )
+        r.loadChunk<int>(shareRaw((int*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::LONG )
+        r.loadChunk<long>(shareRaw((long*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::LONGLONG )
+        r.loadChunk<long long>(shareRaw((long long*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::USHORT )
+        r.loadChunk<unsigned short>(shareRaw((unsigned short*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::UINT )
+        r.loadChunk<unsigned int>(shareRaw((unsigned int*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::ULONG )
+        r.loadChunk<unsigned long>(shareRaw((unsigned long*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::ULONGLONG )
+        r.loadChunk<unsigned long long>(shareRaw((unsigned long long*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::LONG_DOUBLE )
+        r.loadChunk<long double>(shareRaw((long double*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::DOUBLE )
+        r.loadChunk<double>(shareRaw((double*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::FLOAT )
+        r.loadChunk<float>(shareRaw((float*) a.mutable_data()), offset, extent);
+    else if( r.getDatatype() == Datatype::BOOL )
+        r.loadChunk<bool>(shareRaw((bool*) a.mutable_data()), offset, extent);
+    else
+        throw std::runtime_error(std::string("Datatype not known in 'loadChunk'!"));
+
+    return a;
+}
+
+/** Convert a py::typle of py::slices to Offset & Extent
+ *
+ */
+inline std::tuple< Offset, Extent >
+parseTupleSlices(uint8_t const ndim, Extent const & full_extent, py::tuple const & slices) {
+    if( py::len(slices) > ndim )
+        throw std::runtime_error("__getitem__: "
+            "too many indices for dimension of record component!");
+
+    Offset offset(ndim, 0u);
+    Extent extent(ndim, 1u);
+    for( uint8_t i = 0u; i < py::len(slices); ++i )
+    {
+        py::slice slice = py::cast< py::slice >( slices[i] );
```

Comment:
This can be:
- an integer for a single axis
- `np.newaxis` (=`None`)
- an ellipsis `py::ellipsis`
- an empty typle (select all: `[()]`)
- a `py::slice`